uiServices/test/testSuite.feature

```diff
@@ -153,3 +153,183 @@ Feature: Test Suite Creation
+    When I send "PUT" request to "/test-suites/GodogTestSuite" with body:
+  """
+      {
+       "testCases":[
+         {
+         "name":"file1.xml",
+          "preThinkTime": 2000,
+          "postThinkTime": 5000,
+          "execWeight": "Infrequent",
+           "description": "Desc1"
+         }
+        ]
+      }
+    """
+    Then the response code should be 400
+
+
+  Scenario: Successful update of test-suite file with PUT request
+    Given the file "GodogTestSuite.xml" exists at "/uiServices/test/"
+    Given the automated performance ui server is available
+    And the header "testSuitePathDir" is "/uiServices/test/"
+    When I send "PUT" request to "/test-suites/GodogTestSuite" with body:
+    """
+      {
+       "name": "GodogTestSuite2",
+       "testStrategy": "SuiteBased",
+       "description": "ServiceDesc",
+       "testCases":[
+         {
+         "name":"file1.xml",
+          "preThinkTime": 2000,
+          "postThinkTime": 5000,
+          "execWeight": "Infrequent",
+           "description": "Desc1"
+         }
+        ]
+      }
+    """
+    Then the response code should be 204
+    And the response body should be empty
+    When I send a "GET" request to "/test-suites/GodogTestSuite"
+    And the updated file should match json:
+"""
+      {
+       "name": "GodogTestSuite2",
+       "testStrategy": "SuiteBased",
+       "description": "ServiceDesc",
+       "testCases":[
+         {
+         "name":"file1.xml",
+          "preThinkTime": 2000,
+          "postThinkTime": 5000,
+          "execWeight": "Infrequent",
+           "description": "Desc1"
+         }
+        ]
+      }
+    """
+  Scenario: Successful update of test-suite file with PUT request (Update API Name to not match Filename)
+    Given the file "GodogTestSuite.xml" exists at "/uiServices/test/"
+    Given the automated performance ui server is available
+    And the header "testSuitePathDir" is "/uiServices/test/"
+    When I send "PUT" request to "/test-suites/GodogTestSuite" with body:
+    """
+      {
+       "name": "GodogTestSuite2",
+       "testStrategy": "SuiteBased",
+        "description": "ServiceDesc",
+       "testCases":[
+         {
+         "name":"file1.xml",
+          "preThinkTime": 2000,
+          "postThinkTime": 5000,
+          "execWeight": "Infrequent",
+           "description": "Desc1"
+         }
+        ]
+      }
+    """
+    Then the response code should be 204
+    And the response body should be empty
+    When I send a "GET" request to "/test-suites/GodogTestSuite"
+    And the updated file should match json:
+    """
+      {
+       "name": "GodogTestSuite2",
+       "testStrategy": "SuiteBased",
+        "description": "ServiceDesc",
+       "testCases":[
+         {
+         "name":"file1.xml",
+          "preThinkTime": 2000,
+          "postThinkTime": 5000,
+          "execWeight": "Infrequent",
+           "description": "Desc1"
```

Comment:
could be missing something, but do we need description for the test cases at the testSuite level?

Reply:
I have removed it now. Probably wasn't needed at this level.